backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -44,24 +48,32 @@
   private static final String JSON_SECOND_PARTICIPANT_USERNAME = "secondParticipantUsername";
   private static final String JSON_DURATION = "duration";
 
+  // Create GmailFactory and use it to instantiate Email Notifier
+  private final GmailFactory gmFactory = new GmailFactory();
+  private final EmailNotifier emailNotifier = new EmailNotifier(gmFactory.build());
+
+  // Get DatastoreService and instiate Match and Participant Datastores
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final MatchDatastore matchDatastore = new MatchDatastore(datastore);
+  private final ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
+
+  // Get participant username
+  UserService userService = UserServiceFactory.getUserService();
+
+  public SearchMatchServlet() throws GeneralSecurityException, IOException {}
```

Comment:
no longer needed :)

Reply:
i know that constructor sin java are implicit so empty constructors are handled by the JVM but since the class method that is being thrown throws error how are they handled if i remove this constructor or should instead of being exceptions to the build method should try catches be used ?

Reply:
I would recommend having the build method just throw the exception to the caller so the caller can handle them. Either way we should avoid throwing in the constructor.

Reply:
The current build method does throw in the caller ,however, i get a Unhandled exceptions: java.io.IOException, java.security.GeneralSecurityException on line 53 because it is unhandled